### PR TITLE
Add includeTags to /share/id

### PR DESCRIPTION
**Pull request by tobiasKaminsky**

### CODE REVIEW

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 1]
```diff
@@ -379,6 +380,11 @@ public function getShare(string $id): DataResponse {
 		try {
 			if ($this->canAccessShare($share)) {
 				$share = $this->formatShare($share);
+
+				if ($includeTags) {
+					$share = Helper::populateTags($formatted, 'file_source', \OC::$server->getTagManager());
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Controller/ShareAPIController.php:385]:
```suggestion
					$share = Helper::populateTags($share, 'file_source', \OC::$server->getTagManager());
```

File: apps/files_sharing/lib/Controller/ShareAPIController.php [hunk 2]
```diff
@@ -379,6 +380,11 @@ public function getShare(string $id): DataResponse {
 		try {
 			if ($this->canAccessShare($share)) {
 				$share = $this->formatShare($share);
+
+				if ($includeTags) {
+					$share = Helper::populateTags(array($share), 'file_source', \OC::$server->getTagManager());
+				}
+
 				return new DataResponse([$share]);
```
[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:388]: Also no idea. My approach works. 
populateTags needs an array and return one share.

[the remark this answers is not in the document]
Comment (come-nc) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:388]: @nickvergessen seems right, populateTags returns an array, no need to encapsulate it a second time. Should not work even.

Also, `\OC::$server->getTagManager()` is deprecated, use `\OC::$server->get(ITagManager::class)`

[the remark this answers is not in the document]
Comment (CarlSchwan) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:388]: `\OCP\Server::get(ITagManager::class)` is even better

[the remark this answers is not in the document]
Comment (PVince81) [reply apps/files_sharing/lib/Controller/ShareAPIController.php:388]: as you like. changing in the code you touched would be enough I think for now

[the remark this answers is not in the document]
Comment (tobiasKaminsky) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: Is ITagManager really correct?
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/public/ITagManager.php#L42
and
https://github.com/nextcloud/server/blob/215aef3cbdc1963be1bb6bca5218ee0a4b7f1665/lib/private/TagManager.php#L82

I do not see any function I could use?

(I am not that into php, so maybe I understood it wrong, and I do not have a working php IDE…)

[the remark this answers is not in the document]
Comment (PVince81) [reply apps/files_sharing/lib/Controller/ShareAPIController.php]: right, that service is a bit weird/twisted/legacy. let's leave the code as you had it before...

**Review by PVince81 — APPROVED** [review]
:+1: code looks good

**Review by come-nc — APPROVED** [review]

**Pull request merged**